packages/url/src/filter-url-for-display.js

```diff
@@ -19,5 +19,26 @@ export function filterURLForDisplay( url ) {
 		return filteredURL.replace( '/', '' );
 	}
 
+	// Prettify image urls
+	const mediaRegexp = /([\w|:])*\.(?:jpg|jpeg|gif|png|svg)/g;
+	const colon = ':';
+	const period = '.';
+	const query = '?';
```

Comment:
Are these serving much purpose? A few points I'd make:

- In the context of their relevance to this function, they're only relevant _within_ the `if` condition. While it's not much of a problem in this case, in other instances if these assignments were non-trivial function executions, it could cause some wasted resources if that `if` condition is not matched. This is quite similar to what we have enforced in the custom [`no-unused-vars-before-return` ESLint rule](https://github.com/WordPress/gutenberg/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-before-return.md).
- But in any case, to me they appear to fall under the definition of a constant, and as such, would be best to rename and move to the top-most scope ([reference documentation](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#constants)).
- Even still, I'm not clear on what their purpose is. And if it's related to matching ports and query strings, it would be good if we could avoid using string manipulation to work with these values, if we can instead rely on using a [`URL` constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) and one of the properties of its instance ([pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname), etc).
   - Related: #20693

Reply:
Indeed using 
```
let url = new URL('https://developer.mozilla.org/example.mov');
```
will return a nice object with all the elements decomposed like this:
```
hash: ""
host: "developer.mozilla.org"
hostname: "developer.mozilla.org"
href: "https://developer.mozilla.org/example.mov"
origin: "https://developer.mozilla.org"
password: ""
pathname: "/example.mov"
port: ""
protocol: "https:"
search: ""
searchParams: URLSearchParams {  }
username: ""
```
so less need to split/join strings.